tests/core/command/events.js

```diff
@@ -34,6 +34,28 @@
 
 			this.editor.execCommand( 'mockupCommand' );
 			assert.isTrue( cmdCalled, 'Command should be called' );
+		},
+
+		// #17027
+		'test default event data value': function() {
+			var beforeExecData;
+
+			this.editor.once( 'beforeCommandExec', function( evt ) {
+				beforeExecData = evt.data.commandData;
+
+				assert.isObject( beforeExecData, 'Event data is initialized as an empty object.' );
+			}, null, null, 1 );
+
+			this.editor.once( 'afterCommandExec', function( evt ) {
+				assert.areSame( beforeExecData, evt.data.commandData, 'The same object is passed to afterCommandExec.' );
+			}, null, null, 1 );
+
+			this.editor.addCommand( 'dataFlow', {
+				exec: function() {
```

Comment:
Since we're testing `beforeExecData` in `beforeCommandExec` and `afterCommandExec` it would make sense to test it in exec too for the completeness.